opentimelineio/adapters/cmx_3600.py

```diff
@@ -687,123 +687,236 @@ def write_to_string(input_otio, rate=None, style='avid'):
-            url = clip.name
+        clips_and_events += [(child, event)]
 
-        name = clip.name
+    lines = []
 
-        kind = "V" if track.kind == otio.schema.TrackKind.Video else "A"
+    if input_otio.name:
+        lines += ["TITLE: {}".format(input_otio.name), ""]
 
-        lines.append(
-            "{:03d}  {:8} {:5} C        {} {} {} {}".format(
-                edit_number,
-                reel,
-                kind,
-                source_tc_in,
-                source_tc_out,
-                record_tc_in,
-                record_tc_out
+    for clip, event in clips_and_events:
+        lines += [str(event)]
+        if clip:
+            lines += generate_comment_lines(
+                clip,
+                style=style,
+                edl_rate=edl_rate
             )
-        )
-
-        if name:
-            # Avid Media Composer outputs two spaces before the
-            # clip name so we match that.
-            lines.append("* FROM CLIP NAME:  {}".format(name))
-        if url and style == 'avid':
-            lines.append("* FROM CLIP: {}".format(url))
-        if url and style == 'nucoda':
-            lines.append("* FROM FILE: {}".format(url))
-
-        cdl = clip.metadata.get('cdl')
-        if cdl:
-            asc_sop = cdl.get('asc_sop')
-            asc_sat = cdl.get('asc_sat')
-            if asc_sop:
-                lines.append(
-                    "*ASC_SOP ({} {} {}) ({} {} {}) ({} {} {})".format(
-                        asc_sop['slope'][0],
-                        asc_sop['slope'][1],
-                        asc_sop['slope'][2],
-                        asc_sop['offset'][0],
-                        asc_sop['offset'][1],
-                        asc_sop['offset'][2],
-                        asc_sop['power'][0],
-                        asc_sop['power'][1],
-                        asc_sop['power'][2]
-                    ))
-            if asc_sat:
-                lines.append("*ASC_SAT {}".format(
-                    asc_sat
-                ))
 
-        # Output any markers on this clip
-        for marker in clip.markers:
-            timecode = otio.opentime.to_timecode(
-                marker.marked_range.start_time,
-                edl_rate
-            )
+    text = "\n".join(lines) + "\n"
+    return text
 
-            color = marker.color
-            meta = marker.metadata.get("cmx_3600")
-            if not color and meta and meta.get("color"):
-                color = meta.get("color").upper()
-            comment = (marker.name or '').upper()
-            lines.append("* LOC: {} {:7} {}".format(timecode, color, comment))
 
-        # If we are carrying any unhandled CMX 3600 comments on this clip
-        # then output them blindly.
-        extra_comments = clip.metadata.get('cmx_3600', {}).get('comments', [])
-        for comment in extra_comments:
-            lines.append("* {}".format(comment))
+def generate_comment_lines(clip, style, edl_rate, from_or_to='FROM'):
```

Comment:
Top level functions like this should be prefixed with _ to make them hidden, so people outside this module won't accidentally find them (in help, etc.)

Reply:
Sounds great.  Updated.  There are existing top-level functions that don't follow convention and will need similar treatment in the future.

Reply:
If you're willing to _ those too, that'd be great, or I'm happy to do it after this lands.

Reply:
Looks like there was only one more 🎉 .  Updated.